src/pages/DelegatoryValidator/Components/TimeDurationIntervalBar.tsx

```diff
@@ -12,37 +12,55 @@ export default function TimeDurationIntervalBar({
     return null;
   }
 
+  const [remainingTime, setRemainingTime] = useState<moment.Duration>();
+  const [percentageComplete, setPercentageComplete] = useState<number>(0);
+
   // the beginning of the unlock cycle
   const startTime = parseTimestamp(timestamp.toString()).subtract(30, "days");
 
   // the end of the unlock cycle
   const unlockTime = parseTimestamp(timestamp.toString());
 
-  // the remaining time of the unlock cycle
-  const now = moment();
-  const remainingTime = moment.duration(
-    unlockTime.valueOf() - now.valueOf(),
-    "milliseconds",
-  );
+  useEffect(() => {
+    const refresh = () => {
+      // the remaining time of the unlock cycle
+      const now = moment();
+      const remainingTime = moment.duration(
+        unlockTime.valueOf() - now.valueOf(),
+        "milliseconds",
+      );
+      setRemainingTime(remainingTime);
 
-  // the time already passed in the unlock cycle
-  const alreadyPassedTime = moment.duration(
-    now.valueOf() - startTime.valueOf(),
-    "milliseconds",
-  );
+      // the time already passed in the unlock cycle
+      const alreadyPassedTime = moment.duration(
+        now.valueOf() - startTime.valueOf(),
+        "milliseconds",
+      );
+
+      const percentage =
+        (alreadyPassedTime.asMilliseconds() /
+          (unlockTime.valueOf() - startTime.valueOf())) *
+        100;
+      setPercentageComplete(percentage);
+    };
+
+    refresh();
+    setInterval(refresh, 60 * 1000); // refresh every minute
+
+    const remainingTimeInMS = remainingTime?.asMilliseconds();
+
+    if (remainingTimeInMS !== undefined && remainingTimeInMS <= 0) {
+      window.location.reload();
+    }
+  }, [remainingTime, percentageComplete]);
```

Comment:
do we want to update a component every minute? I think we can simply do
```
const MINUTE_MS = 60000;

useEffect(() => {
  const interval = setInterval(() => {
    console.log('Logs every minute');
  }, MINUTE_MS);

  return () => clearInterval(interval); // This represents the unmount function, in which you need to clear your interval to prevent memory leaks.
}, [])
```